Assets/Scripts/UI/ItemButton.cs

```diff
@@ -0,0 +1,28 @@
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace UI
+{
+    public class ItemButton : MonoBehaviour
+    {
+        [SerializeField] private Image icon;
+        [SerializeField] private TextMeshProUGUI nameLabel;
+        [SerializeField] private TextMeshProUGUI countLabel;
+
+        private int _count;
+
+        public Image Icon => icon;
+        public TextMeshProUGUI NameLabel => nameLabel;
+
+        public int Count
```

Comment:
Add summary, so we know what it is and that it updates the text automatically.
Other than that:
![](https://media.giphy.com/media/8Odq0zzKM596g/giphy.gif)